### PR TITLE
Enable the spdlog recipe to use the bundled version of the fmt library

### PR DESCRIPTION
**Pull request by bvanessen**
that ships with its repo.

### CODE REVIEW

**Comment by bvanessen** [comment]
@alalazo Do you have time to look at this one?

File: var/spack/repos/builtin/packages/spdlog/package.py [hunk 1]
```diff
@@ -59,7 +64,7 @@ def cmake_args(self):
             args.extend(
                 [
                     self.define_from_variant("SPDLOG_BUILD_SHARED", "shared"),
-                    self.define("SPDLOG_FMT_EXTERNAL", "ON"),
+                    self.define("SPDLOG_FMT_EXTERNAL", ("~bundled_fmt" in self.spec)),
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/spdlog/package.py]: Can we use a variant that has the same logic as this option? In this way we can use `define_from_variant`

**Review by alalazo — CHANGES_REQUESTED** [review]
Only a minor request.

File: var/spack/repos/builtin/packages/spdlog/package.py (hunk 1 above)
Comment (bvanessen) [reply var/spack/repos/builtin/packages/spdlog/package.py]: Done.

**Review by alalazo — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Pull request merged**